src/transaction/id.rs

```diff
@@ -84,6 +49,11 @@ impl Transaction {
 
             Output::Change { amount, .. } => *amount = 0,
 
+            Output::Message { recipient, amount } => {
```

Comment:
is this duplicating `prepare_init`?

Reply:
Its a different context. Even tho we do the same, here its related to the signature, not to the VM initialization.

Reply:
Makes sense, clearing for signatures isn't the same as for init (eg witnesses etc)